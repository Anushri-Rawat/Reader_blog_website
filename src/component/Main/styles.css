.title {
  margin: 24px 0 0;
  padding-bottom: 5px;
  font-size: 20px;
  font-weight: bolder;
}
.articles-container {
  overflow-y: scroll;
  height: 100vh;
  padding-right: 2rem;
}
.btn:hover {
  background: white !important;
  color: #6c757d !important;
}
@media (max-width: 425px) {
  .title {
    margin: 10px 0 0;
  }
  .articles-container {
    padding-right: 0;
  }
}
