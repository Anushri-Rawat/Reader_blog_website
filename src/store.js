import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import posts from "./reducers";

const initialState = {
  loading: false,
  posts: JSON.parse(localStorage.getItem("posts")) || [],
};

const store = () =>
  createStore(posts, initialState, composeWithDevTools(applyMiddleware(thunk)));

export default store;
