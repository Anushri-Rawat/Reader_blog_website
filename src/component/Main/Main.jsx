import React, { useState } from "react";
import { useSelector } from "react-redux";
import SearchBar from "../SearchBar/SearchBar";
import BlogCard from "../BlogCard/BlogCard";
import Form from "../Form/Form";
import "./styles.css";

const Main = () => {
  const { posts } = useSelector((state) => state);
  const [searchQuery, setSearchQuery] = useState("");
  let listToDisplay = posts;

  if (searchQuery !== "") {
    listToDisplay = posts.filter((item) => {
      return (
        item.title.toLowerCase().includes(searchQuery.toLocaleLowerCase()) ||
        item.body.toLowerCase().includes(searchQuery.toLocaleLowerCase())
      );
    });
  }

  return (
    <main style={{ padding: "1.8% 3%" }} className="d-flex" data-testid="main">
      <div style={{ width: "100%" }}>
        <SearchBar setSearchQuery={setSearchQuery} />
        <div className="border-bottom title">Articles</div>
        <div className="articles-container">
          {listToDisplay.length > 0 ? (
            listToDisplay?.map((item, i) => <BlogCard key={i} data={item} />)
          ) : (
            <p>No result found!!</p>
          )}
        </div>
      </div>
      <div
        className="border-bottom d-none d-md-block"
        style={{ minWidth: "350px", paddingLeft: "1.5rem" }}
      >
        <Form />
      </div>
    </main>
  );
};

export default Main;
