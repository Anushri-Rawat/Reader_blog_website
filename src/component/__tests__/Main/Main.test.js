import { render as rtlRender, screen, cleanup } from "@testing-library/react";
import store from "../../../store";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import Main from "../../Main/Main";
// import { fetchAllPosts } from "../../../actions";

// jest.mock("../../../actions");

afterEach(() => {
  cleanup();
});

const render = (component) =>
  rtlRender(<Provider store={store()}>{component}</Provider>);

test("should render Main component", () => {
  render(<Main />);
  const mainElement = screen.getByTestId("main");
  expect(mainElement).toBeInTheDocument();
});

test("should render searchbar(child) component", () => {
  const { getByTestId } = render(<Main />);
  expect(getByTestId("search-bar")).toBeInTheDocument();
});

test("should render form(child) component", () => {
  const { getByTestId } = render(<Main />);
  expect(getByTestId("form")).toBeInTheDocument();
});

// test("should render BlogCard(child) component", async () => {
//   fetchAllPosts.mockResolvedValueOnce({ listToDisplay: posts });
//   const { getByTestId } = render(<Main />);
//   await wait(() => expect(getByTestId("card")).toBeInTheDocument());
//   expect(fetchAllPosts).toHaveBeenCalledOnce();
// });

test("matches snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store()}>
        <Main />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
