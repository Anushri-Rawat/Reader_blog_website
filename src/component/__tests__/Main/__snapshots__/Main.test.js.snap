// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<main
  className="d-flex"
  data-testid="main"
  style={
    Object {
      "padding": "1.8% 3%",
    }
  }
>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="d-flex justify-content-between align-items-center searchBar-container"
      data-testid="search-bar"
    >
      <input
        className="search-bar"
        onChange={[Function]}
        placeholder="Search"
        type="search"
      />
      <select
        className="sort-container"
        onChange={[Function]}
      >
        <option
          value="id"
        >
          Id
        </option>
        <option
          value="title"
        >
          Title
        </option>
        <option
          value="userId"
        >
          User Id
        </option>
      </select>
    </div>
    <div
      className="border-bottom title"
    >
      Articles
    </div>
    <div
      className="articles-container"
    >
      <p>
        No result found!!
      </p>
    </div>
  </div>
  <div
    className="border-bottom d-none d-md-block"
    style={
      Object {
        "minWidth": "350px",
        "paddingLeft": "1.5rem",
      }
    }
  >
    <form
      className="d-flex flex-column gap-3"
      data-testid="form"
      onSubmit={[Function]}
    >
      <h2
        className="title mt-0"
      >
        Add Blog
      </h2>
      <div
        className="row align-items-center"
      >
        <div
          className="col-md-3"
        >
          <label
            htmlFor="userId"
          >
            User Id
          </label>
        </div>
        <div
          className="col-md-9"
        >
          <select
            className="form-input"
            id="userId"
            name="UserId"
            onChange={[Function]}
          >
            <option
              value={1}
            >
              1
            </option>
            <option
              value={2}
            >
              2
            </option>
            <option
              value={3}
            >
              3
            </option>
            <option
              value={4}
            >
              4
            </option>
            <option
              value={5}
            >
              5
            </option>
            <option
              value={6}
            >
              6
            </option>
            <option
              value={7}
            >
              7
            </option>
            <option
              value={8}
            >
              8
            </option>
            <option
              value={9}
            >
              9
            </option>
            <option
              value={10}
            >
              10
            </option>
          </select>
        </div>
      </div>
      <div
        className="row align-items-center"
      >
        <div
          className="col-md-3"
        >
          <label
            htmlFor="title"
          >
            Title
          </label>
        </div>
        <div
          className="col-md-9"
        >
          <input
            className="form-input"
            id="title"
            onChange={[Function]}
            placeholder="Title"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="row align-items-center"
      >
        <div
          className="col-md-3"
        >
          <label
            htmlFor="body"
          >
            Body
          </label>
        </div>
        <div
          className="col-md-9"
        >
          <textarea
            className="form-input"
            id="body"
            onChange={[Function]}
            placeholder="Body"
            rows={5}
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="btn btn-dark rounded-0"
      >
        Add Blog
      </button>
    </form>
  </div>
</main>
`;
